include/recognition/Cograph/FactorizingPermutation.hpp

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <list>
+#include <graph/GraphTools.hpp>
+
+#include "Part.hpp"
+
+namespace Koala {
+    class FactorizingPermutation {
+    public:
+        part *l, *r, *first_part, *last_part;
+        element *origin;
+        std::vector<element *> E;
```

Comment:
In general would it be a problem to convert it to an automatically managed piece of memory?